salt/metalk8s/volumes/prepared/init.sls

```diff
@@ -0,0 +1,47 @@
+include:
+  - .installed
+
+{%- set volumes_to_create = [] %}
+{%- set all_volumes = pillar.metalk8s.volumes %}
+{%- set target_volume = pillar.get('volume') %}
+
+{#- If a volume is given we only take this one, otherwise we take them all. #}
+{%- if target_volume is not none %}
+  {%- if target_volume in all_volumes.keys() %}
+    {%- do volumes_to_create.append(target_volume) %}
+  {%- else %}
+Volume "{{ target_volume }}" not found in pillar:
+  test.fail_without_changes: []
+  {%- endif %}
+{%- else %}
+    {%- do volumes_to_create.extend(all_volumes.keys()|list) %}
+{%- endif %}
+
+Create the sparse file spool:
```

Comment:
`spool`? `pool`?

Reply:
I meant `spool` but after checking it seems I had a wrong definition of this word so I'll use `directory` instead (let's keep things simple).

Reply:
Done